### ISSUE TITLE
[Question]: Does VidGear support downloading arbitrary frames and getting video sizes?

### ISSUE THREAD

**Issue by slerman12** [abhiTronix/vidgear#381]
### Issue guidelines

- [X] I've read the [Issue Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/issue/#submitting-an-issue-guidelines) and wholeheartedly agree.

### Issue Checklist

- [X] I have searched open or closed [issues](https://github.com/abhiTronix/vidgear/issues) for my problem and found nothing related or helpful.
- [X] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest) and found nothing related to my problem.
- [X] I have gone through the [Bonus Examples](https://abhitronix.github.io/vidgear/latest/help/get_help/#bonus-examples) and [FAQs](https://abhitronix.github.io/vidgear/latest/help/get_help/#frequently-asked-questions) and found nothing related or helpful.

### Describe your Question

`CamGear(source=url).start()` and `.read()` allow me to read frames sequentially. Is it possible to access a frame by index? Furthermore, can I get the size (number of frames in the video) a priori?

Thanks.

### VidGear Version

0.3.2

### Python version

3.11

### Operating System version

MacOS Ventura

### Any other Relevant Information?

No.

**Comment by abhiTronix** [comment]
@slerman12 what's the source? A camera, a URL or something else. Kindly elaborate your setup.

**Comment by slerman12** [comment]
Never mind. Just a YouTube video. Ordinary YouTube video. But I need to get it frame-wise by index. 

My solution right now is to iterate via `video.read()` up to the desired frame index. However, there's no `video.restart()` option to start the iteration back from the first index.

`video.stop()` followed by `video.start()` doesn't seem to reset the video stream index back to 0 and video.read()returns None after it has iterated through the whole video once.

In other words, I'm looking for a `video.restart()` option (without reinitializing the whole `CamGear(source=url, stream_mode=True)`).

**Comment by abhiTronix** [comment]
@slerman12 If you need accurate seeking and frames with index use my [DeFFcode](https://github.com/abhiTronix/deffcode) library, example can be found here: https://github.com/abhiTronix/deffcode/issues/34#issuecomment-1288803403
